rocketpy/stochastic/stochastic_flight.py

```diff
@@ -63,11 +67,18 @@ def __init__(
         terminate_on_apogee : bool, optional
             Whether or not the flight should terminate on apogee. This attribute
             can not be randomized.
+        time_overshoot : bool
+            If False, the simulation will run at the time step defined by the controller
+            sampling rate. Be aware that this will make the simulation run much slower.
         """
         if terminate_on_apogee is not None:
             assert isinstance(terminate_on_apogee, bool), (
                 "`terminate_on_apogee` must be a boolean"
             )
+        if time_overshoot is not None:
+            assert isinstance(time_overshoot, bool), (
+                "`time_overshoot` must be a boolean"
+            )
```

Comment:
Better not use `assert` in production codes (https://oandersonbm.medium.com/python-dont-use-assert-like-if-statement-b63b0a7e25b3).

```suggestion
        if time_overshoot is not None:
            if not isinstance(time_overshoot, bool):
                raise TypeError("`time_overshoot` must be a boolean")
```

I know you just copied from elsewhere, but it is a known issue that we have `assert` spread throughout the code. We will try to remove them in the future.